skyportal/handlers/api/obj.py

```diff
@@ -0,0 +1,68 @@
+from sqlalchemy.orm import joinedload
+from baselayer.app.access import auth_or_token, AccessError
+from baselayer.app.env import load_env
+from ..base import BaseHandler
+from ...models import DBSession, Obj
+
+_, cfg = load_env()
+
+
+class ObjHandler(BaseHandler):
+    @auth_or_token  # ACLs will be checked below based on configs
+    def delete(self, obj_id):
+        """
+        ---
+        description: Delete an Obj
+        tags:
+          - objs
+        parameters:
+          - in: path
+            name: obj_id
+            required: true
+            schema:
+              type: string
+        responses:
+          200:
+            content:
+              application/json:
+                schema: Success
+          400:
+            content:
+              application/json:
+                schema: Error
+        """
+        obj = Obj.get_if_accessible_by(
+            obj_id,
+            self.current_user,
+            mode='delete',
+            raise_if_none=True,
+            # Load in all related entities into the session
+            # so that they can be checked by the permission verification
+            options=[
+                joinedload(Obj.photometry),
+                joinedload(Obj.spectra),
+                joinedload(Obj.comments),
```

Comment:
Others may have commented but the user could still own all other data, i.e. photometry, spectra, etc

Reply:
Ah yeah, I wasn't sure what to do with that. I'm fine with allowing that if you are.

Reply:
Yeah, I think comments should be excluded from this check. And perhaps classifications? Maybe these are questions we should bring to others in the group (the astronomers, specifically)

Reply:
Yeah I think that makes sense. I agree this is probably something we should ask the astronomers for confirmation on. @guynir42 @dmitryduev thoughts? Should users be able to delete an Obj even if not all comments/classifications on it were submitted by themselves?

Reply:
I think this is ok in principle, but it does mean we will have to implement the patch as well as the delete methods. 

I think that if someone is sure that he wants to delete an object, then the conversation should go with it. 

If the poster just wants to make changes to the object it should be through patch, but if that is not available then he will have to delete and repost. Then we could end up losing comments (and annotations!) for no good reason. 

I hope I can get on this when I settle in at Berkeley but I don't know when exactly that would happen...

Reply:
Yeah, needing a PATCH does make sense. Looks like we do expose that functionality as the Source PATCH handler already [here](https://skyportal.io/docs/api.html#tag/sources/paths/~1api~1sources~1obj_id/patch). Is there anything missing functionality-wise from that that users would need from a Obj PATCH handler, do you think?

Reply:
Ahhh there's a source PATCH. Great. I think that probably answers most of the use-cases. Except if someone had posted a bunch of un-saved objs that are not considered sources. But that seems like a fairly small edge case that we can handle when we do the full handlers overhaul.